packages/accounts-controller/src/AccountsController.test.ts

```diff
@@ -2403,6 +2408,29 @@ describe('AccountsController', () => {
       ).toBe('new name');
     });
 
+    it('sets the nameLastUpdatedAt timestamp when setting the name of an existing account', () => {
+      const expectedTimestamp = Number(new Date('2024-01-02'));
+
+      jest.spyOn(Date, 'now').mockImplementation(() => expectedTimestamp);
+
+      const { accountsController } = setupAccountsController({
+        initialState: {
+          internalAccounts: {
+            accounts: { [mockAccount.id]: mockAccount },
+            selectedAccount: mockAccount.id,
+          },
+        },
+      });
+      accountsController.setAccountName(mockAccount.id, 'new name');
+
+      expect(
+        accountsController.getAccountExpect(mockAccount.id).metadata
+          .nameLastUpdatedAt,
+      ).toBe(expectedTimestamp);
+
+      jest.spyOn(Date, 'now').mockRestore();
```

Comment:
I feel like the "spy" could "bleed out" of this test scope if the previous `expect` fails...

I believe we would need to adds a `afterEach(() => { // clear all mocks }` (but since we don't have it now, it might also undo global mocks, which would have other side effects...)

Maybe for now, we could live with a `mockImplementationOnce` instead? so that we won't need to restore it here?

Reply:
Thanks for doing the change!

Just to be extra clear (if anyone re-read this later), mocking once could steal "bleed out" on the next text if this test never reaches the `Date.now` while renaming the account.

But this is unlikely to happen, or that means that something else has been broken. So I think this will be enough for now 👍